Source/DOH/file.c

```diff
@@ -286,21 +286,36 @@ DOH *DohNewFile(DOHString *filename, const char *mode, DOHList *newfiles) {
   char *filen;
 
   filen = Char(filename);
+
   file = fopen(filen, mode);
-  if (!file)
-    return 0;
+  if (!file) {
+    return NULL;
+  }
 
   f = (DohFile *) DohMalloc(sizeof(DohFile));
+  if (!f) {
+    fclose(file);
+    return NULL;
+  }
+
   if (newfiles)
     Append(newfiles, filename);
+
   f->filep = file;
   f->fd = 0;
   f->closeondel = 1;
   obj = DohObjMalloc(&DohFileType, f);
+  if (!obj) {
+    DohFree(f);
+    fclose(file);
+    return NULL;
+  }
```

Comment:
Same applies to `DohObjMalloc()` (this exit-on-allocation-failure strategy was implemented in e38847f7e161c022f794de6bf28d5f7da5cba3c1 which debuted in SWIG 4.1.0 - maybe you looked at the code of an older release?)